adding language analyzers (#8591)
opensearch-project/documentation-website: _analyzers/language-analyzers/arabic.md

---
layout: default
title: Arabic
parent: Language analyzers
grand_parent: Analyzers
nav_order: 10
---

# Arabic analyzer

The built-in `arabic` analyzer can be applied to a text field using the following command:

```json
PUT /arabic-index
{
  "mappings": {
    "properties": {
      "content": {
        "type": "text",
        "analyzer": "arabic"
      }
    }
  }
}
```
{% include copy-curl.html %}

## Stem exclusion

> You can also use `stem_exclusion` with this language analyzer using the following command:

Reviewer: ```suggestion
You can use `stem_exclusion` with this language analyzer using the following command:
```

```json
PUT index_with_stem_exclusion_arabic
{
  "settings": {
    "analysis": {
      "analyzer": {
        "stem_exclusion_arabic_analyzer":{
          "type":"arabic",
          "stem_exclusion":["تكنولوجيا","سلطة "]
        }
      }
    }
  }
}
```
{% include copy-curl.html %}

## Arabic analyzer internals

> The `arabic` analyzer is build using the following:

Reviewer: ```suggestion
The `arabic` analyzer is built using the following components:
```

> The `arabic` analyzer is build using the following:

Reviewer: Global: please apply these changes to all files.

> Tokenizer: `standard`

Reviewer: ```suggestion
- Tokenizer: `standard`
```

> Token filters:

Reviewer: ```suggestion
- Token filters:
```

> - lowercase

Reviewer: Could you make lines 56-61 a sublist?

  - decimal_digit
  - stop (Arabic)
  - normalization (Arabic)

>   - keywords

Reviewer 2: ```suggestion
  - keyword
```

- stemmer (Arabic)

## Custom Arabic analyzer

> You can create custom Arabic analyzer using the following command:

Reviewer: ```suggestion
You can create a custom Arabic analyzer using the following command:
```
